R/coh_plot.R

```diff
@@ -0,0 +1,324 @@
+#' the graphic elements.To do so follow the convention:
+#' c("c1" = vaccinated_color, "c2" = unvaccinated_color)
+#' @param data dataset with cohort information (see example)
+#' @param outcome_status_col name of the column containing status of the
+#' event (most be a binary column)
+#' @param time_to_event_col name of the column containing the time-to-event
+#' @param vacc_status_col name of the column containing the vaccination
+#' status
+#' @param vaccinated_status string assigned to the vaccinated population
+#' @param unvaccinated_status string assigned to the unvaccinated population
+#' @param vaccinated_color color assigned to the vaccinated population
+#' @param unvaccinated_color color assigned to the unvaccinated population
+#' @param start_cohort start date of the study
+#' @param end_cohort end date of the study
+#' @param percentage if TRUE returns probability in percentage
+#' @param cumulative if TRUE returns cumulative Hazards (1-Survival)
+#' @return ggplot2 plot of survival/cumulative hazard
+#' @examples
+#' # load example package data
+#' data("cohortdata")
+#' head(cohortdata)
+#' start_cohort <- as.Date("2044-01-01")
+#' end_cohort <- as.Date("2044-12-31")
+#'
+#' cohortdata$immunization_death <-
+#'   get_immunization_date(
+#'     data = cohortdata,
+#'     outcome_date_col = "death_date",
+#'     outcome_delay = 0,
+#'     immunization_delay = 14,
+#'     vacc_date_col = c("vaccine_date_1", "vaccine_date_2"),
+#'     end_cohort = end_cohort,
+#'     take_first = FALSE
+#'   )
+#'
+#' cohortdata$vaccine_status <- set_status(
+#'   data = cohortdata,
+#'   col_names = "immunization_death",
+#'   status = c("v", "u")
+#' )
+#'
+#' cohortdata$death_status <- set_status(
+#'   data = cohortdata,
+#'   col_names = "death_date"
+#' )
+#'
+#' cohortdata$time_to_death <- get_time_to_event(
+#'   data = cohortdata,
+#'   outcome_date_col = "death_date",
+#'   start_cohort = start_cohort,
+#'   end_cohort = end_cohort,
+#'   start_from_immunization = FALSE
+#' )
+#'
+#' plot_survival(data = cohortdata,
+#'   outcome_status_col = "death_status",
+#'   time_to_event_col = "time_to_death",
+#'   vacc_status_col = "vaccine_status",
+#'   vaccinated_status = "v",
+#'   unvaccinated_status = "u",
+#'   vaccinated_color = "steelblue",
+#'   unvaccinated_color = "darkred",
+#'   start_cohort = start_cohort,
+#'   end_cohort = end_cohort,
+#'   percentage = TRUE,
+#'   cumulative = TRUE
+#' )
+#' @export
+plot_survival <- function(data, outcome_status_col,
+                          time_to_event_col,
+                          vacc_status_col,
+                          vaccinated_status = "v",
+                          unvaccinated_status = "u",
+                          vaccinated_color = "steelblue",
+                          unvaccinated_color = "darkred",
+                          start_cohort,
+                          end_cohort,
+                          percentage = TRUE,
+                          cumulative = FALSE) {
```

Comment:
1. The plot() method returns a very limited plot. There are other libraries specialized in plotting survival outputs, but including an additional dependence only for this plot is too much.

Reply:
Okay agreed, this seems to be a decent compromise for now. One consideration would be to take the {incidence2} approach and place {ggplot2} in Suggests, to avoid taking it on as a 'full' dependency. You would need to add a check for whether {ggplot2} is installed in the `plot_*()` functions, but it's a reasonable assumption that many users will have it. See how {incidence2} does it: https://github.com/reconverse/incidence2/blob/ed6dc9f49e107879a229be05dee408eb9147da4a/R/plot.R#L147